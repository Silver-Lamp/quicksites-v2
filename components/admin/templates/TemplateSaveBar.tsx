import { Button } from '../ui/button';
import { Dispatch, SetStateAction, useState } from 'react';
import { supabase } from '@/admin/lib/supabaseClient';
import toast from 'react-hot-toast';

type TemplateSaveBarProps = {
  template: any;
  rawJson: string;
  commitMessage: string;
  setSaveStatus: Dispatch<SetStateAction<'idle' | 'saved' | 'error'>>;
  setHighlightErrors: Dispatch<SetStateAction<boolean>>;
};

export default function TemplateSaveBar({
  template,
  rawJson,
  commitMessage,
  setSaveStatus,
  setHighlightErrors
}: TemplateSaveBarProps) {
  const [saveStatus, setLocalSaveStatus] = useState<'idle' | 'saved' | 'error'>('idle');

  const handleSave = async () => {
    if (!template?.template_name || !template?.industry || !template?.layout || !template?.data?.pages?.length) {
      setHighlightErrors(true);
      return;
    }

    try {
      const { data: user } = await supabase.auth.getUser();
      const editor_id = user?.user?.email || 'unknown';
      const payload = { ...template, data: JSON.parse(rawJson) };

      const res = await fetch('/api/templates/update', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          ...payload,
          commit_message: commitMessage,
          editor_id
        })
      });

      if (res.ok) {
        setLocalSaveStatus('saved');
        setSaveStatus('saved');
        toast.success('Template saved successfully.');
        setTimeout(() => {
          setLocalSaveStatus('idle');
          setSaveStatus('idle');
        }, 3000);
      } else {
        setLocalSaveStatus('error');
        setSaveStatus('error');
        toast.error('Failed to save template.');
      }
    } catch (err) {
      setLocalSaveStatus('error');
      setSaveStatus('error');
      toast.error('Failed to save template.');
    }
  };

  return (
    <div className="flex justify-between items-center fixed bottom-0 left-0 right-0 bg-background border-t p-4">
      <div className="flex items-center gap-2">
        <Button onClick={handleSave}>Save Changes</Button>
        {saveStatus === 'saved' && (
          <span className="text-green-500 text-xs animate-pulse">✓ Saved</span>
        )}
        {saveStatus === 'error' && (
          <span className="text-red-500 text-xs">⚠️ Error</span>
        )}
      </div>
    </div>
  );
}
