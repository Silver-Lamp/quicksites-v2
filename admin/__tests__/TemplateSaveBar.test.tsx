import { render } from '@testing-library/react';
import TemplateSaveBar from '../components/templates/TemplateSaveBar';

it('matches snapshot', () => {
  const { asFragment } = render(
    <TemplateSaveBar jsonView={true} toggleView={() => {}} onSave={() => {}} saveStatus="idle" />
  );
  expect(asFragment()).toMatchSnapshot();
});
