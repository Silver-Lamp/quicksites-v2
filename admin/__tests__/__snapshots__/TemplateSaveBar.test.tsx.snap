// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between items-center"
  >
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80 h-9 px-4 py-2"
    >
      Toggle Form View
    </button>
    <div
      class="flex items-center gap-2"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
      >
        Save Changes
      </button>
    </div>
  </div>
</DocumentFragment>
`;
